source/MaterialXGenShader/Shader.h

```diff
@@ -281,8 +288,8 @@ class Shader
     ShaderGraphPtr _rootGraph;
     vector<ShaderGraph*> _graphStack;
 
-    size_t _activeStage;
-    vector<Stage> _stages;
+    Stage* _activeStage;
```

Comment:
Curious why a StagePtr is not just kept here since a shared pointer is kept in the map. Seems safer to me as well.

Reply:
It's my performance paranoia :D 

This pointer is dereferenced a lot during shader gen, and at least in debug builds the shared_ptr's dereference operatator-> gets called as a function, instead of direct access as with a raw pointer.. I guess in release builds this is probably not a performance issue as it is probably not a function call in that case. So might be a moot point in terms of performance.

But it's also annoying when debugging code having to go in and out of this dereference function call. So that's why I made it a raw pointer. I'm not worried about safety as the same class has ownership of the shared_ptr that owns the instance (they are kept safe in the _stages map).

Reply:
Okay. Sounds reasonable.